bitcask/compaction.go

```
@@ -0,0 +1,282 @@
+	var compactedFiles []int
+
+	var currentFile *os.File
+	var currentFileID int
+	var currentEntryCount int
+
+	startNewCompactedFile := func() error {
+		if currentFile != nil {
+			if err := currentFile.Sync(); err != nil {
+				return err
+			}
+			currentFile.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `currentFile.Close` is not checked

(errcheck)